extensions/amp-story-dev-tools/0.1/amp-story-dev-tools.js

```diff
@@ -32,21 +36,182 @@ const fontsToLoad = [
   },
 ];
 
+/**
+ * Generates the template for the root layout.
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildContainerTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-dev-tools-container">
+      <div class="i-amphtml-story-dev-tools-header">
+        <span class="i-amphtml-story-dev-tools-brand">
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            width="30"
+            height="30"
+            viewBox="0 0 30 30"
+            fill="none"
+          >
+            <circle cx="15" cy="15" r="15" fill="white"></circle>
+            <path
+              fill="#202125"
+              d="M19.5 9a1.5 1.5 0 011.5 1.5v9a1.5 1.5 0 01-1.5 1.5V9zM8.25 9a1.5 1.5 0 011.5-1.5h6.75A1.5 1.5 0 0118 9v12a1.5 1.5 0 01-1.5 1.5H9.75a1.5 1.5 0 01-1.5-1.5V9zM22.5 10.5c.621 0 1.125.504 1.125 1.125v6.75c0 .622-.504 1.125-1.125 1.125v-9z"
+            />
+          </svg>
+          <span class="i-amphtml-story-dev-tools-brand-text">
+            <span>WEB STORIES</span>
+            <span>DEV - TOOLS</span>
+          </span>
+        </span>
+        <div class="i-amphtml-story-dev-tools-tabs"></div>
+        <a
+          class="i-amphtml-story-dev-tools-button i-amphtml-story-dev-tools-close"
+        >
+          <span>OPEN STORY</span>
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            width="14"
+            height="14"
+            viewBox="0 0 14 14"
+            fill="none"
+          >
+            <path
+              d="M9.9165 4.08333L9.094 4.90583L10.599 6.41667H4.6665V7.58333H10.599L9.094 9.08833L9.9165 9.91667L12.8332 7L9.9165 4.08333ZM2.33317 2.91667H6.99984V1.75H2.33317C1.6915 1.75 1.1665 2.275 1.1665 2.91667V11.0833C1.1665 11.725 1.6915 12.25 2.33317 12.25H6.99984V11.0833H2.33317V2.91667Z"
+              fill="black"
+            />
+          </svg>
+        </a>
+      </div>
+    </div>
+  `;
+};
+
+/**
+ * Updates the hashString with the dictionary<string, string> passed in
+ * @public
+ * @param {!Object} updates
+ * @param {!Window} win
+ */
+export function updateHash(updates, win) {
+  let queryHash = parseQueryString(win.location.hash);
+  queryHash = Object.assign(queryHash, updates);
+  win.location.hash = Object.entries(queryHash)
+    .filter((e) => e[1] != undefined)
```

Comment:
Nit: you can add this test to the `map` below and save one full cycle over the loop. Not that it matters but 🤷

Reply:
Not sure what you mean by this. What I want to do here is to pass `{tab: undefined}` when the tab is the devices preview, and it should remove the tab attribute from the hashString (because it's the default). How would it be possible to filter from the `map`? I'd rather not have `&&` in hashStrings, just one `&` between the key-value pairs